src/cats/labs/channel.cljc

```diff
@@ -78,64 +64,47 @@
 
     p/Functor
     (-fmap [_ f mv]
-      (let [c (a/chan 1 (map f))]
-        (a/pipe mv c)
-        c))
+      (a/pipe mv (a/chan 1 (map f))))
 
     p/Semigroup
     (-mappend [_ sv sv']
-      (chain-chans sv sv'))
+      (a/merge [sv sv']))
```

Comment:
Like many sets, there are probably many options for which operation to choose from to create a Monoid.

The three laws are left identity, right identity, and associativity (https://en.wikipedia.org/wiki/Monoid#Definition).
Associativity is: `a <> (b <> c) = (a <> b) <> c` (where <> represents an infix mappend)

merge satisfies the identity laws when mempty is an empty, closed channel. It also satisfies the associativity law under this form of equality `(defn chan-eq [c d] (= (<!! (a/into #{} c)) (<!! (a/into #{} d))))`.

If you think of channels as being equal only if they are ordered the same, then no, merge does not satisfy the associativity law. But channels are the binding of time and data, and since time is inherently unpredictable in computing, I think it is wrong to include ordering (or time) when defining channel equality. So I would define channel equality to be the set-based method above, and therefore merge would satisfy the laws.

I looked around for some other instances of streams/channels having monoid instances but didn't find many:
- [scalaz.StreamT](https://github.com/scalaz/scalaz/blob/v7.2.0/core/src/main/scala/scalaz/StreamT.scala#L71) seems to agree with the `chain-chans` method 
- [Haskell's pipes](https://hackage.haskell.org/package/pipes-4.1.7/docs/src/Pipes-Internal.html#line-116) performs an `mappend` on the inner values.

I think all three approaches can be correct under some equality operator. But perhaps the best of both worlds is the inner mappend like the pipes library uses. That would allow ordering to be preserved and infinite streams to be handled

A rough sketch:
```clojure
(defn mappend-chans [c d]
  (go-loop []
    (let [vc (<! c) vd (<! d)]
      (when (and vc vd)
        (>! out (mappend vc vd))
        (recur)))))

(<!! (a/into [] (m/append (a/to-chan [[1]]) (a/to-chan [[2] [3]])))) ; => [[1 2]]
```